curated_transformers/models/scalar_weight.py

```diff
@@ -0,0 +1,147 @@
+from io import BytesIO
+from typing import Any, List
+
+import torch
+from torch.nn import Module
+from torch import Tensor
+import srsly
+
+from spacy.util import SimpleFrozenDict
+from thinc.model import Model
+from thinc.types import ArgsKwargs, Ragged
+from thinc.layers.pytorchwrapper import PyTorchWrapper_v2
+from thinc.shims.pytorch_grad_scaler import PyTorchGradScaler
+from thinc.util import torch2xp, xp2torch
+
+from ..util import all_equal
+
+# From syntaxdot:
+# https://github.com/tensordot/syntaxdot/blob/22bd3d43ed2d7fcbef8a6217b01684194fae713f/syntaxdot-transformers/src/scalar_weighting.rs#L62
+class ScalarWeight(Module):
+    def __init__(self, *, num_layers: int, dropout_prob: float = 0.1):
+        super().__init__()
+
+        self.layer_weights = torch.nn.parameter.Parameter(
+            torch.zeros((num_layers), requires_grad=True)
+        )
+        self.scale = torch.nn.parameter.Parameter(
+            torch.tensor((1.0,), requires_grad=True)
+        )
+        self.dropout = torch.nn.Dropout(p=dropout_prob)
```

Comment:
I think we can just store the probability, since the dropout layer itself is never used?